[receiver/mongodb] Add Scaffolding for MongoDB receiver (#6972)
open-telemetry/opentelemetry-collector-contrib: receiver/mongodbreceiver/README.md

# Mongodb Receiver

This receiver fetches stats from a Mongodb instance using the [golang
mongo driver](https://github.com/mongodb/mongo-go-driver). Stats are collected
via MongoDB's `dbStats` command.

Supported pipeline types: `metrics`

> :construction: This receiver is in **BETA**. Configuration fields and metric data model are subject to change.

## Prerequisites

This receiver supports MongoDB versions:

- 2.6
- 3.0+
- 4.0+
- 5.0

## Configuration

The following settings are required:

> - `hosts` (default: [`localhost:27017`]): list of `host:port` or unix domain socket endpoints.

Reviewer: Since it has a default, this isn't required.

  - For standalone MongoDB deployments this is the hostname and port of the mongod instance

>   - For replica sets specify the hostnames and ports of the mongod instances that are in the replica set configuration. Another requirement is that the `replica_set` field must also be set with the name of the replica set for autodiscovery to work.

Reviewer: ```suggestion
  - For replica sets specify the hostnames and ports of the mongod instances that are in the replica set configuration. If the `replica_set` field is specified, nodes will be autodiscovered.
```

Can you reword this? Maybe something like this?

>   - For replica sets specify the hostnames and ports of the mongod instances that are in the replica set configuration. Another requirement is that the `replica_set` field must also be set with the name of the replica set for autodiscovery to work.

Reviewer: If `replica_set` is specified, does the user only need to specify one mongod, or is it necessary to specify all of them?

Author: From my understanding, it is expected to specify the mongod instances which are specified in the configuration of the replica set. 

https://docs.mongodb.com/manual/reference/connection-string/#replica-set-option

  - For a sharded MongoDB deployment, please specify a list of the `mongos` hosts.

The following settings are optional:

- `username`: If authentication is required, the user can be provided here.
- `password`: If authentication is required, the password can be provided here.
- `collection_interval` (default = `10s`): This receiver collects metrics on an interval. This value must be a string readable by Golang's [time.ParseDuration](https://pkg.go.dev/time#ParseDuration). Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.

> - `replica_set`: If the deployment of MongoDB is a replica set then this allows users to specify the replica set name which allows for autodiscovery of other nodes in the replica set.

Reviewer: What happens if this is specified when the deployment is not a replica set?

Author: Did a little more digging and the mongo driver won't know that its not a replica set and will fail to connect. With what I have currently spiked out it would give this error:
```
Error: cannot start receivers: unable to validate connection: unable to connect: unable to instantiate mongo client: could not connect to hosts [localhost:27017]: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [] }
```
